river/drift/retrain.py

```diff
@@ -39,10 +41,22 @@ class DriftRetrainingClassifier(base.Wrapper, base.Classifier):
 
     """
 
-    def __init__(self, model: base.Classifier, drift_detector: base.DriftAndWarningDetector = None):
+    def __init__(
+        self,
+        model: base.Classifier,
+        drift_detector: base.DriftAndWarningDetector = None,
+        use_bkg: bool = True,
+    ):
         self.model = model
         self.bkg_model = model.clone()
+        self.use_bkg = use_bkg
         self.drift_detector = drift_detector if drift_detector is not None else drift.DDM()
+        if self.use_bkg:
+            self.bkg_model = model.clone()
+            if not isinstance(self.drift_detector, base.DriftAndWarningDetector):
+                raise Exception(
+                    "Drift detection method must have a warning tracker if parameter use_bkg is True"
+                )
 
```

Comment:
No need to check this. When the `warning_detected` property is checked in `learn_one`, an exception will be raised if necessary.